metaflow/datastore/datastore.py

```diff
@@ -525,7 +525,8 @@ def _register_data_artifacts(self, objects, info):
                                   ds_type=self.TYPE,
                                   url=self.object_path(sha),
                                   sha=sha,
-                                  type=info[var]['encoding'])
+                                  type=info[var]['encoding'],
+                                  size=info[var]['size'])
```

Comment:
I believe the size that is in info corresponds to the size of the pickled object and I believe you want (which I agree with) to return the size of the unpickled object. You may have to modify _save_object in datastore.py to properly return the unpickled size.

Reply:
@romain-intel Yeah, I made mistake in the return size of _save_object. And I have modified this unpickled object size. Please review this again.